### ISSUE TITLE
Out of memory error with simple (4 coordinates) case

### ISSUE THREAD

**Issue by MikkoSteerpath** [mapbox/polylabel#106]
```
const polygon = [[[0,0],[1.1203426968636618,5.994245193525291e-15],[1.2133824496332195,0],[0,0]]]
const precision = 0.5;

polylabel(polygon, precision)
```

Version used 1.1.0

## Expected result

Some coordinate close to [0.6, 0] or even an exception to be thrown.

## Actual result

Quickly eating all memory until out of memory error and process is killed.

**Comment by Richard004** [comment]
I see. The problem is pretty obvious.

If your input contains polygon like this.

    [10,0],[0,0],[10.1,6.0E-09],[5.8E-06,6.2E-09]

Where width to height proportion is some 1e-9
Then the `// cover polygon with initial cells` for-cycles virtually try to generate 1e9 instances to queue. 
My fix (C#) was to check 

    var polySize = Math.Max(width, height);
    var cellSize = Math.Min(width, height);

    if (cellSize == 0) 
		return (minX, minY);	// clearly zero polygon

    if (cellSize / polySize < 1e-3)
                return ((maxX + minX) / 2, (maxY + minY) / 2);  degenerated flat polygon, give up and return cca midle of the box

**Comment by Richard004** [comment]
Here I am saying that some order 1000 instances are ok (that is the 1e-3 cell size to poly size ratio) but for more I give up and do not run the algorithm at all.

**Comment by mourner** [comment]
This should be fixed by cc66e38c0d514e44f5d62964a173074e8a227baa
